test/lib/ufe/testDeleteCmd.py

```diff
@@ -310,10 +421,240 @@ def testDeleteArgs(self):
+        trunkPath = ufe.Path([mayaPathSegment, usdPathSegment])
+        trunkItem = ufe.Hierarchy.createItem(trunkPath)
+
+        ufe.GlobalSelection.get().append(trunkItem)
+
+        # get the USD stage
+        stage = mayaUsd.ufe.getStage(str(mayaPathSegment))
+
+        # by default edit target is set to the Rootlayer.
+        self.assertEqual(stage.GetEditTarget().GetLayer(), stage.GetRootLayer())
+
+        # delete selected item `/TreeBase/trunk` `
+        cmds.delete()
+        # deleted two items
+        self.assertEqual(ufeObs.nbDeleteNotif() , 2)
+
+    @unittest.skipUnless(ufeUtils.ufeFeatureSetVersion() >= 2, 'testDeleteProxyShape only available in UFE v2 or greater.') 
+    def testDeleteRestrictionVariant(self):
+        '''Deleting prims inside a variantSet is not allowed.'''
+
+        cmds.file(new=True, force=True)
+
+        # open Variant.ma scene in testSamples
+        mayaUtils.openVariantSetScene()
+
+        # Create our UFE notification observer
+        ufeObs = TestObserver()
+        ufe.Scene.addObserver(ufeObs)
+
+        # stage
+        mayaPathSegment = mayaUtils.createUfePathSegment('|Variant_usd|Variant_usdShape')
+        stage = mayaUsd.ufe.getStage(str(mayaPathSegment))
+
+        # first check that we have a VariantSets
+        objectPrim = stage.GetPrimAtPath('/objects')
+        self.assertTrue(objectPrim.HasVariantSets())
+
+        # Geom
+        usdPathSegment = usdUtils.createUfePathSegment('/objects/Geom')
+        geomPath = ufe.Path([mayaPathSegment, usdPathSegment])
+        geomItem = ufe.Hierarchy.createItem(geomPath)
+        geomPrim = mayaUsd.ufe.ufePathToPrim(ufe.PathString.string(geomPath))
+
+        # Small_Potato
+        smallPotatoUsdPathSegment = usdUtils.createUfePathSegment('/objects/Geom/Small_Potato')
+        smallPotatoPath = ufe.Path([mayaPathSegment, smallPotatoUsdPathSegment])
+        smallPotatoItem = ufe.Hierarchy.createItem(smallPotatoPath)
+        smallPotatoPrim = mayaUsd.ufe.ufePathToPrim(ufe.PathString.string(smallPotatoPath))
+
+        # add Geom to selection list
+        ufe.GlobalSelection.get().append(geomItem)
+
+        # get prim spec for Geom prim
+        primspecGeom = stage.GetEditTarget().GetPrimSpecForScenePath(geomPrim.GetPath());
+
+        # primSpec is expected to be None
+        self.assertIsNone(primspecGeom)
+
+        cmds.delete()
+        # expect no delete
+        self.assertEqual(ufeObs.nbDeleteNotif() , 0)
+
+        # clear selection
+        cmds.select(clear=True)
+
+        # add Small_Potato to selection list 
+        ufe.GlobalSelection.get().append(smallPotatoItem)
+
+        # get prim spec for Small_Potato prim
+        primspecSmallPotato = stage.GetEditTarget().GetPrimSpecForScenePath(smallPotatoPrim.GetPath())
+
+        # primSpec is expected to be None
+        self.assertIsNone(primspecSmallPotato)
+
+        # delete "/objects/Geom/Small_Potato"
+        # expect no delete
+        cmds.delete()
```

Comment:
What's the difference between this test and the one at line 626?

Reply:
Its checking that the object under geom couldnt be removed as well